### PR TITLE
[Synapse] az synapse: Add for managed private endpoints commands

### PR DESCRIPTION
**Pull request by kevinzz6**
**Description**

Add synapse managed private endpoints commands. Included create, show, list and delete managed-private-endpoints.
Add azure-synapse-managedprivateendpoints =0.3.0
**Testing Guide**

**History Notes**

[Synapse] az synapse: Add managed private endpoints feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Synapse

File: src/azure-cli/azure/cli/command_modules/synapse/commands.py [hunk 1]
```diff
@@ -478,5 +482,13 @@ def get_custom_sdk(custom_module, client_factory):
         g.custom_command('export', 'export_notebook')
         g.custom_command('delete', 'delete_notebook', confirmation=True, supports_no_wait=True)
 
+    # Data Plane Commands --Managed private endpoints operations
+    with self.command_group('synapse managed-private-endpoints', synapse_managed_private_endpoints_sdk,
+                            custom_command_type=get_custom_sdk('managedprivateendpoints', None)) as g:
+        g.custom_show_command('show', 'get_Managed_private_endpoints')
+        g.custom_command('create', 'create_Managed_private_endpoints', supports_no_wait=True)
+        g.custom_command('list', 'list_Managed_private_endpoints')
+        g.custom_command('delete', 'delete_Managed_private_endpoints', confirmation=True, supports_no_wait=True)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/synapse/commands.py:491]: Both `create` and `delete` support no wait but there seems no `wait` command

File: src/azure-cli/azure/cli/command_modules/synapse/tests/latest/test_synapse_scenario.py [hunk 2]
```diff
@@ -1362,3 +1362,49 @@ def test_integration_runtime(self):
...
+    #@record_only()
+    @ResourceGroupPreparer(name_prefix='synapse-cli', random_name_length=16)
+    def test_managed_private_endpoints(self):
+        self.kwargs.update({
+            'workspace': 'testsynapseworkspacepe',
+            'name': 'AzureDataLakeStoragePE',
+            'privateLinkResourceId': '/subscriptions/051ddeca-1ed6-4d8b-ba6f-1ff561e5f3b3/resourceGroups/bigdataqa/providers/Microsoft.Storage/storageAccounts/hozhao0917gen2',
+            'groupId': 'dfs'})
+        
+        #self._create_workspace("--enable-managed-virtual-network")
```
Comment (idear1203) [inline src/azure-cli/azure/cli/command_modules/synapse/tests/latest/test_synapse_scenario.py:1375]: Remove this line?

File: src/azure-cli/azure/cli/command_modules/synapse/tests/latest/test_synapse_scenario.py [hunk 3]
```diff
@@ -1362,3 +1362,49 @@ def test_integration_runtime(self):
             checks=[
                 self.check('name', self.kwargs['selfhosted-integration-runtime'])
             ])
+
+    #@record_only()
```
Comment (idear1203) [inline src/azure-cli/azure/cli/command_modules/synapse/tests/latest/test_synapse_scenario.py:1366]: Remove this line?

File: .gitignore [hunk 4]
```diff
@@ -6,6 +6,7 @@ __pycache__/
 env/
 env27/
 venv/
+env1/
```
Comment (idear1203) [inline .gitignore:9]: Pleas don't change this file

**Review by idear1203 — CHANGES_REQUESTED** [review]

File: src/azure-cli/azure/cli/command_modules/synapse/commands.py (hunk 1 above)
Comment (kevinzz6) [reply src/azure-cli/azure/cli/command_modules/synapse/commands.py]: api doesn't suport wait command, so delete supports_no_wait=True.

**Comment by kevinzz6** [comment]
@evelyn-ys  @idear1203  please take time to review my new commit, thanks.

**Review by jsntcy — APPROVED** [review]

**Pull request merged**